### PR TITLE
(GH-8503) Add about_Module_Manifests

### LINKED ISSUE #8503: Add/Update sample manifest to show all fields and possible values
### Prerequisites

- [X] Write a descriptive title.
- [X] Search the existing issues.
- [X] I am reporting the documentation problem for version of PowerShell I am using.

### Version

5.1, 7.0, 7.1, 7.2, 7.3 (preview)

### Link to affected document

https://docs.microsoft.com/en-us/powershell/scripting/developer/module/how-to-write-a-powershell-module-manifest?view=powershell-7.2#sample-module-manifest
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/new-modulemanifest?view=powershell-7.2#examples

### Description of the documentation error

The sample manifest has a number of commented out properties and it would be useful to have a sample manifest that showed how each could be used.

### Suggested fix

Add a new sample or update existing one (happy for this to be assigned to me to fix)

### PR DESCRIPTION
**Pull request by michaeltlombardi**
# PR Summary
Adds the `about_Module_Manifests` topic.

- Resolves #8503 

## PR Context

Check the boxes below to indicate the content affected by this PR.

**Repository or docset configuration**
- [ ] Repo documentation and configuration (.git/.github/.vscode etc.)
- [ ] Docs build files (.openpublishing.* and build scripts)
- [ ] Docset configuration (docfx.json, mapping, bread, module folder)

**Conceptual documentation**
- [ ] Files in docs-conceptual

**Cmdlet reference & about_ topics**
When changing **cmdlet reference** or **about_ topics**, the changes should be copied to all
relevant versions. Check the boxes below to indicate the versions affected by this change.

- [x] Preview content
- [x] Version 7.2 content
- [x] Version 7.1 content
- [x] Version 7.0 content
- [x] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**. If the PR is work in progress,
  please add the prefix `WIP:` or `[WIP]` to the beginning of the title and remove the prefix when
  the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md [hunk 1]
```diff
@@ -0,0 +1,1714 @@
...
+- a `.ps1` or `.psm1` file makes the module type **Script**
+- a `.psd1` file makes the module type **Manifest**
+- a `.dll` file makes the module type **Binary**
+- a `.cdxml` file makes the module type **CIM**
+- a `.xaml` file makes the module type **Workflow**
+
+To export members from `.psm1` or `.dll` files in a module that has a manifest,
+the names of those files must be specified in the values of the **RootModule**
+or **NestedModules** keys in the manifest. Otherwise, their members aren't
+exported.
```
Comment (sdwheeler) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md:128]: What about cdxml and xaml types, are their values exported differently? In script-based modules, you can use the `Export-ModuleMember` cmdlet.

So really, you mean that for exports defined in the manifest, the files containing the items to be exported must be listed in **RootModule** or **NestModules**.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md [hunk 2]
```diff
@@ -0,0 +1,1714 @@
...
+#### When RootModule is a relative path
+
+In this example, PowerShell loads `Example.psm1` from the `Files` folder
+relative to the `.psd1` manifest file.
+
+```powershell
+@{
+    RootModule = 'Files/Example.psm1'
+}
+```
```
Comment (sdwheeler) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md:176]: I am not sure that these are needed. It seems redundant to the bullet list above.

**Review by sdwheeler — CHANGES_REQUESTED** [review]
Commit my review comments and taking a break. I will add more comments later.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md [hunk 3]
```diff
@@ -0,0 +1,1714 @@
...
+#### When CompatiblePSEditions is set to Desktop and Core
+
+This module can only be imported in sessions where the `$PSEdition` automatic
+variable exists.
+
+```powershell
+@{
+    CompatiblePSEditions = @('Desktop', 'Core')
+}
+```
```
Comment (sdwheeler) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md:296]: Ditch the H4s. Make the description more imperative. Don't need to show all 4 possible combinations. Just explain that "Undefined is the same as `@('Desktop', 'Core')`.

An empty array is allowed. Do we know if that is the same as both? Or what does it mean to be 'none'?

**Review by sdwheeler — CHANGES_REQUESTED** [review]
more comments

**Review by sdwheeler — CHANGES_REQUESTED** [review]
Looking much better. Try to separate consecutive alert blocks.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md [hunk 4]
```diff
@@ -0,0 +1,1476 @@
...
+> [!NOTE]
+> `RequiredVersion` was added in Windows PowerShell 5.0.
+> `MaximumVersion` was added in Windows PowerShell 5.1.
+
+Typically, nested modules contain commands that the root module needs for its
+internal processing. By default, the commands in nested modules are exported
+from the module's session state into the caller's session state, but the root
+module can restrict the commands that it exports with the `Export-ModuleMember`
+command. You can always use the **FunctionsToExport** and **CmdletsToExport**
+settings to further restrict exported commands.
```
Comment (sdwheeler) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md:916]:
```suggestion
Any items that need to be exported from a nested module must exported by the
nested module using the `Export-ModuleMember` cmdlet or be listed in one of the
export properties:

- **FunctionsToExport**
- **CmdletsToExport**
- **VariablesToExport**
- **AliasesToExport**
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md [hunk 5]
```diff
@@ -0,0 +1,1476 @@
...
+
+|                       |       Value       |
+| --------------------- | ----------------- |
+| **Input Type**        | `System.String[]` |
+| **Required**          | No                |
+| **Value if unset**    | `$null`           |
+| **Accepts wildcards** | Yes               |
+
+You can specify entries in this setting with wildcards. All matching functions
+in the list of exported functions is exported.
```
Comment (sdwheeler) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md:952]:
```suggestion
in the list of exported functions are exported.
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md [hunk 6]
```diff
@@ -0,0 +1,1476 @@
...
+
+> [!TIP]
+> For performance and discoverability, you should always explicitly list the
+> functions you want your module to export in this setting without using any
+> wildcards.
+
+For example, when you import this module, all functions the root module and any
+nested modules export are available.
```
Comment (sdwheeler) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md:960]:
```suggestion
For example, when you import a module with the setting commented out, 
all functions in the root module and any nested modules are exported.
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md [hunk 7]
```diff
@@ -0,0 +1,1476 @@
...
+You can specify entries in this setting with wildcards. All matching cmdlets
+in the list of exported cmdlets is exported.
+
+> [!TIP]
+> For performance and discoverability, you should always explicitly list the
+> cmdlets you want your module to export in this setting without using any
+> wildcards.
+
+For example, when you import this module, all cmdlets the root module and any
+nested modules export are available.
```
Comment (sdwheeler) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md:1039]:
```suggestion
For example, when you import a module with this setting commented out, 
all cmdlets in the root module and any nested modules are exported.
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md [hunk 8]
```diff
@@ -0,0 +1,1476 @@
...
+You can specify entries in this setting with wildcards. All matching variables
+in the list of exported module members is exported.
+
+> [!TIP]
+> For performance and discoverability, you should always explicitly list the
+> variables you want your module to export in this setting without using any
+> wildcards.
+
+For example, when you import this module, all variables the root module and any
+nested modules export are available.
```
Comment (sdwheeler) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md:1118]:
```suggestion
For example, when you import a module with this setting commented out,
all variables in the root module and any nested modules are exported.
```

**Review by sdwheeler — CHANGES_REQUESTED** [review]

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md [hunk 9]
```diff
@@ -0,0 +1,1476 @@
...
+```
+
+### Description
+
+This setting describes the module at a high level.
+
+|                       |         Value          |
+| --------------------- | ---------------------- |
+| **Input Type**        | `System.String`        |
+| **Required**          | For PowerShell Gallery |
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md:347]:
```suggestion
| **Required**          | PowerShell Gallery |
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md [hunk 10]
```diff
@@ -0,0 +1,1476 @@
...
+
+### PrivateData
+
+This setting defines a hash table of data which is passed to the defined
+**RootModule** on import.
+
+|                       |             Value              |
+| --------------------- | ------------------------------ |
+| **Input Type**        | `System.Collections.Hashtable` |
+| **Required**          | For PowerShell Gallery         |
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Module_Manifest.md:1342]:
```suggestion
| **Required**          | PowerShell Gallery         |
```

**Review by sdwheeler — APPROVED** [review]

**Pull request merged**